docs/walkthrough/walkthrough-istio.md

```diff
@@ -0,0 +1,384 @@
+* [Installing the istio operator](#installing-the-istio-operator)
+* [Installing the istio service mesh](#installing-the-istio-service-mesh)
+* [Note on sidecar injection](#note-on-sidecar-injection)
+* [Installing Tekton Pipelines](#installing-tekton-pipelines)
+* [Installing Tekton Triggers (optional)](#installing-tekton-triggers-optional)
+* [Installing Tekton Dashboard](#installing-tekton-dashboard)
+* [Setting up the istio Gateway and VirtualService](#setting-up-the-istio-gateway-and-virtualservice)
+* [Browsing the istio dashboard](#browse-the-istio-dashboard)
+* [Adding basic authentication](#adding-basic-authentication)
+* [Cleaning up](#cleaning-up)
+
+## Before you begin
+
+Before you begin, make sure the following tools are installed:
+
+1. [`kind`](https://kind.sigs.k8s.io): For creating a local cluster running on top of docker.
+1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl): For interacting with your kubernetes cluster.
+1. [`istioctl`](https://istio.io/latest/docs/setup/getting-started/#download): For installing istio.
+
+Nothing more is required to run the walkthrough !
+
+## Service mesh benefits
+
+Using a service mesh brings benefits in several domains:
+- security
+- observability
+- resiliency
+
+Some security benefits:
+- mutual TLS brings secure communications inside the cluster (encrypted and authenticated)
+- egress control prevents data exfiltration and domains white list
+- network policies allow to secure pod-to-pod or service-to-service communication at the network and application layers
+
+Some observality benefits:
+- understand how service performance impacts things upstream and downstream
+- help detect and fix issues quickly and efficiently
+
+Some resiliency benefits:
+- automatically retry failed service calls
+- mirror traffic to send a copy of live traffic to a mirrored service
+- add circuit breakers directly on the wire
+- inject network faults to test applications behaviour
+
+The list is by no means exhaustive and service meshes bring a lot more benefits than just those enumerated above.
+
+Visit [istio's webpage](https://istio.io/latest/docs/concepts/what-is-istio/) or [istio's blog](https://istio.io/latest/blog) for more info.
+
+## Creating a local Kubernetes cluster
+
+The first thing to do is to create a Kubernetes cluster.
+
+Kind is an easy solution to run a local cluster, all it needs is to have `docker` installed.
+
+Create a cluster by running the following command:
+
+```bash
+kind create cluster --name walkthrough --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraPortMappings:
+  - containerPort: 30080
+    hostPort: 80
+    protocol: TCP
+  - containerPort: 30443
+    hostPort: 443
+    protocol: TCP
+EOF
+```
+
+This will create a cluster with ports 80 and 443 exposed on the host computer. This is needed to communicate with the istio ingress gateway.
```

Comment:
```suggestion
This will create a cluster with ports 80 and 443 exposed on the host computer. This is needed to communicate with the Istio ingress gateway.
```